transmission/transmission.go

```diff
@@ -246,10 +246,17 @@ func (b *batchAgg) fireBatch(events []*Event) {
 	if numEncoded == 0 {
 		return
 	}
-	// get some attributes common to this entire batch up front
-	apiHost := events[0].APIHost
-	writeKey := events[0].APIKey
-	dataset := events[0].Dataset
+	// get some attributes common to this entire batch up front off the first
+	// valid event (some may be nil)
+	var apiHost, writeKey, dataset string
+	for _, ev := range events {
```

Comment:
probably not a huge deal if batch sizes are kept small, but this is another O(n) loop.  any chance we can do this while in `encodeBatch`?  returning three additional strings seems pretty lame, but maybe encodeBatch could return an `*Event`?

Reply:
Since it will early-out 99.9% of the time on the first iteration, I don't see that the time complexity is a problem.

Reply:
It's only o(n) when all events but the last have failed to be encoded. That seems like a pathalogically bad case no matter what. For the common case, this loop will early exit after examining the first event. And even if it does have to run through a bunch of them, walking a list of pointers with nothing but a `nil` check will be super fast.

I did look at trying to return an `*Event` with an empty payload and it looked like it would make every other check throughout encoding super messy.